content/how-to-use-supertokens-custom-ui-with-vuejs/index.md

```diff
@@ -0,0 +1,608 @@
+
+Here are the functions for Github and Google respectively:
+
+```vue
+...
+      onGithubPressed: async function () {
+          const authUrl = await ThirdPartyEmailPassword.getAuthorisationURLWithQueryParamsAndSetState({
+              providerId: "github",
+
+              // This is where github should redirect the user back after login or error.
+              // This URL goes on the github dashboard as well.
+              authorisationURL: `http://localhost:3000/auth/callback/github`,
+          });
+
+          window.location.assign(authUrl);
+      },
+      onGooglePressed: async function () {
+          const authUrl = await ThirdPartyEmailPassword.getAuthorisationURLWithQueryParamsAndSetState({
+              providerId: "google",
+
+              // This is where google should redirect the user back after login or error.
+              // This URL goes on the google dashboard as well.
+              authorisationURL: `${websiteDomain}/auth/callback/google`,
+          });
+
+          window.location.assign(authUrl);
+      },
+ ……
+```
+
+The `getAuthorisationURLWithQueryParamsAndSetState` method returns the callback URL that takes us to the `/auth/callback` page that we discussed above.
+
+After social authentication is completed and the user is redirected to the `/auth/callback/<provider>` route, we call the `thirdPartySignInAndUp` function from `supertokens-web-js` which consumes the authorisation code to sign in the user.
+
+Here is the function that handles the above flow in the `AuthCallbackView` component inside `/src/views/AuthCallbackView.vue` file:
+
+```vue
+  mounted: async function () {
+       try {
+      
+           const response = await ThirdPartyEmailPassword.thirdPartySignInAndUp({});
+           if (response.status !== "OK") {
+               return window.location.assign("/auth?error=signin");
+           }
+           // sign in successful. The session tokens are handled
+           // automatically via our SDK.
+           window.location.assign("/");
+       } catch (_) {
+              window.location.assign("/auth?error=signin");
+       }
+   }
+
+```
+
+In the background, we show a small loading animation on the `/auth/callback/<providerId>` page using the HTML template [here](https://github.com/supertokens/supertokens-web-js/blob/master/examples/vuejs/with-thirdpartyemailpassword/src/html/authCallbackView.html). 
+
+### Setup Routing to Show the Signup/Login forms
+
+Vue CLI already generates the initial routing for our app inside `/src/router.index.ts`. 
+
+We’ll update this file so that `/auth` route loads the `AuthView` component and the `/auth/callback/:provider` route loads the `AuthCallbackView` component we created earlier: 
```

Comment:
```suggestion
We'll update this file so that the `/auth` route loads the `AuthView` component and the `/auth/callback/:provider` route loads the `AuthCallbackView` component we created earlier:
```